### PR TITLE
Fix bug: Include comment for @param tag with nested tag

### PR DESCRIPTION
**Pull request by andy-ms**
Fixes https://github.com/Microsoft/TypeScript/issues/23221#issuecomment-379812189

`parseChildParameterOrPropertyTag` contains code to skip ahead to the next line -- this means that the comment goes missing. We should probably *never* skip a comment, but that would be a bigger change than this.

### CODE REVIEW

File: src/compiler/parser.ts [hunk 1]
```diff
@@ -6526,6 +6526,8 @@ namespace ts {
                     const result = target === PropertyLikeParse.Parameter ?
                         <JSDocParameterTag>createNode(SyntaxKind.JSDocParameterTag, atToken.pos) :
                         <JSDocPropertyTag>createNode(SyntaxKind.JSDocPropertyTag, atToken.pos);
+                    let comment: string | undefined;
+                    if (indent !== undefined) comment = parseTagComments(indent + scanner.getStartPos() - atToken.pos);
```
Comment (sandersn) [inline src/compiler/parser.ts]: If only nested `@param`s were missing comments before this change, where was the comment parsed for non-nested ones? Does this change now parse the comments for all `@param`s here?

File: src/harness/unittests/jsDocParsing.ts [hunk 2]
```diff
@@ -305,6 +305,12 @@ namespace ts {
  * @param x hi
 < > still part of the previous comment
  */`);
+
+                parsesCorrectly("Nested @param tags",
+`/**
+* @param {object} o Doc doc
+* @param {string} o.f
```
Comment (sandersn) [inline src/harness/unittests/jsDocParsing.ts]: I thought it was the comments on nested `@param` that were incorrectly skipped. Shouldn’t this test include some comments for this line too?

Comment (andy-ms) [reply src/harness/unittests/jsDocParsing.ts]: `parseChildParameterOrPropertyTag` *begins* by skipping a line. So it's the comment that comes *before* the child that was skipped. But I'll add some comments for that line too.

File: src/compiler/parser.ts (hunk 1 above)
Comment (andy-ms) [reply src/compiler/parser.ts]: Yeah, previously it was always done at the bottom of `parseTag`. But that didn't work if the comment had already been skipped.

**Pull request merged**